### ISSUE TITLE
bug in dijkstra_shortest_paths / closest_vertices

### ISSUE THREAD

**Issue by sbromberger** [JuliaGraphs/LightGraphs.jl#705]
MWE:
```
g = SimpleWeightedGraph(CycleGraph(4))
add_edge!(g, 1, 4, 2)
dijkstra_shortest_paths(g, 2, allpaths=true, trackvertices=true).closest_vertices
# returns [2 1 3 4 4]
```

**Comment by sbromberger** [comment]
cc @somil55 :)

**Comment by sbromberger** [comment]
Do you think the fix is as easy as just truncating `closest_vertices` before returning from `dijkstra`?

**Comment by somil55** [comment]
@sbromberger Checkout my PR on weighted betweenness centrality. The fix is already there.

**Comment by sbromberger** [comment]
Resolved in #706.
